backend/app/api/endpoints/recommendations.py

```
@@ -0,0 +1,130 @@
+            founder_id=current_user.id,
+            max_results=max_results,
+            min_score=min_score,
+        )
+
+        return recommendations
+
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to generate recommendations: {str(e)}",
+        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)